Remove ProcessGroup from TensorPipeAgent initialization

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#66708 Remove ProcessGroup from TensorPipeAgent initialization**
* #66257 Formatting changes
* #65946 Remove process group barrier and all_reduce function calls from tensorpipe agent



cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Differential Revision: [D31762735](https://our.internmc.facebook.com/intern/diff/D31762735)

H-Huang committed:
Remove ProcessGroup from TensorPipeAgent initialization

[ghstack-poisoned]

H-Huang committed:
Update on "Remove ProcessGroup from TensorPipeAgent initialization"




cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

mrshenli reviewed (approved):
LGTM! Thanks! Might need to rebase to get rid of the test failures.

H-Huang committed:
Update on "Remove ProcessGroup from TensorPipeAgent initialization"




cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Differential Revision: [D31762735](https://our.internmc.facebook.com/intern/diff/D31762735)

[ghstack-poisoned]

H-Huang committed:
Update on "Remove ProcessGroup from TensorPipeAgent initialization"




cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Differential Revision: [D31762735](https://our.internmc.facebook.com/intern/diff/D31762735)

[ghstack-poisoned]

H-Huang committed:
Update on "Remove ProcessGroup from TensorPipeAgent initialization"




cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Differential Revision: [D31762735](https://our.internmc.facebook.com/intern/diff/D31762735)

[ghstack-poisoned]

H-Huang committed:
Update on "Remove ProcessGroup from TensorPipeAgent initialization"




cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

Differential Revision: [D31762735](https://our.internmc.facebook.com/intern/diff/D31762735)

[ghstack-poisoned]
